autoload/arduino.vim

```diff
@@ -96,6 +100,17 @@ function! arduino#GetArduinoExecutable() abort
   endif
 endfunction
 
+function! arduino#GetBuildPath() abort 
+  if empty(g:arduino_build_path)
+    return ''
+  endif
+  let l:path = g:arduino_build_path
+  let l:path = substitute(l:path, '{file}', expand('%:p'), 'g')
+  let l:path = substitute(l:path, '{project_dir}', fnamemodify(expand('%:p'), ":h"), 'g')
```

Comment:
You can just use `expand('%:p:h')`

Reply:
Uh great, thanks!
Rebased and changed it.